spylon/spark/yarn_launcher.py

```diff
@@ -313,9 +321,18 @@ def launcher(deploy_mode, args, working_dir="."):
         run_pyspark_yarn_cluster(**args)
 
 
+def _fix_permissions(env_dir):
+    bin_dir = os.path.join(env_dir, "bin")
+    if os.path.isdir(bin_dir):
+        if platform.system() == 'Linux':
+            subprocess.check_call(["chmod", "-R", "a+x", bin_dir])
```

Comment:
If this call fails, will the output be captured on the user's machine or will this be swallowed by the cluster?

Essentially I want to make sure that the output of this command and show it to the user to aid in debugging.

Example of failed call:

```
In [19]: ret1 = subprocess.check_call("chmod -R a+x /tmp/foo".split())
chmod: changing permissions of '/tmp/foo/bar': Operation not permitted    <========
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
<ipython-input-19-263ebda4d87d> in <module>()
----> 1 ret1 = subprocess.check_call("chmod -R a+x /tmp/foo".split())

/home/eric/miniconda/lib/python3.5/subprocess.py in check_call(*popenargs, **kwargs)
    579         if cmd is None:
    580             cmd = popenargs[0]
--> 581         raise CalledProcessError(retcode, cmd)
    582     return 0
    583 

CalledProcessError: Command '['chmod', '-R', 'a+x', '/tmp/foo']' returned non-zero exit status 1
```

I am especially interested in making sure that the user can see the line that I marked with `    <========`

Would you consider replacing this `check_call` with the following code snippet that can capture the stdout/stderr of the call?

```python
p = subprocess.Popen(cmd, stderr=subprocess.PIPE, 
                 stdout=subprocess.PIPE)
stdout, stderr = p.communicate()

if p.returncode != 0:
    raise RuntimeError("Command: {} failed. returncode: {}, "
                       "stdout: {}, stderr: {}"
                       "".format(cmd, returncode, stdout, 
                                 stderr, p.returncode))
```